### PR TITLE
fix(rx-utils): fix createHandler signature

### LINKED ISSUE #183: new problem with create-handler
createHandler has problem with union types

### PR DESCRIPTION
**Pull request by scink**
had to delete `THandle` and `THandler` types because TS forms union type incorrectly for the handler. it is `(a: true) => void | (a: false) => void` instead of `(a: boolean) => void`.
fixes #183 

http://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgCoAs4gCYBsIBqA9sNsgN4BQAkABRwD8AXMgG4nYCUL7pA3JQC+lUJFiIUqABJY8hOLgCuEADwBBAHwUa9Fmu5sOA4QHoTyCAA8ADkShhkYAJ7XJMnPnXIAvIdJbfNQtLSBwAZz8yBmR6ZkjOHy1eMhZdZH1EyIErW3tHFzdZTyDfZIC0TA9CDmQAHzR3OQIFZXUNAUpsCARcOCgUBCIQMIc4Fmki1TKBQeHRsZiFkahQAHME7ySa3zg+ZDMLKCg7SlmR5DgAI1SNrdIfC72DsPQiRVwyS5QwxQQkMLCpyG50QqQWihwEBgoAgXEyyQeu325hebw+yC+yB+fwgAMonW6vX6yDODmuDUmKmWa3aQLmGIWaWpIHW8O2GKeKNe70+31+-0BpIx5Not0iD0unMOxygdPOlwQYJYEK60JAsLFCN8kuR0pOBJ6fQGwIciopVRmJpJjKWYBWLM17IQUugMrlppFjvuvmdutRPIxfJxeKFfyVyBVUJhcM24p9Uv96Mx2IF+K6huJQuw40ankuRCI+CwtKzNpYzNZsa1yGwLqOJyznrZ3prdbdWbNaUjao1zbIvlrutdDat2GzixY+cLECwXv7rb93KTQYFQA

BREAKING CHANGE: remove THandle and THandler types

### CODE REVIEW

File: packages/rx-utils/src/create-handler.utils.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
 import { Observable, Subject } from 'rxjs';
 
-export interface THandle<A> {
-	(): void;
-	(a: A): void;
+interface HandlerVoid {
+	readonly value$: Observable<void>;
+	readonly handle: (a?: void) => void;
```
Comment (raveclassic) [inline packages/rx-utils/src/create-handler.utils.ts]: What does `(a?: void)` control and what for?

Comment (scink) [reply packages/rx-utils/src/create-handler.utils.ts]: to save possibility to pass undefined as an argument. a little compatibility with the old code. but if it is a breaking change we could force not to pass anything.

Comment (raveclassic) [reply packages/rx-utils/src/create-handler.utils.ts]: Yeah let's keep things clean

Comment (scink) [reply packages/rx-utils/src/create-handler.utils.ts]: amended it

File: packages/rx-utils/src/__tests__/create-handler.utils.spec.ts [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
 describe('create-handler.utils.ts', function() {
 	describe('createHandler', function() {
 		it('should not fail', function() {
 			const a = createHandler();
 			const af1: Lazy<void> = a.handle; // adds possibility to use handle as Lazy<void>
 			const af2: Endomorphism<void> = a.handle; // keeps possibility to use as Endomorphism<void>
 			a.handle();
-			a.handle(undefined);
-			a.handle(constVoid());
+			// a.handle(undefined); // error
```
Comment (raveclassic) [inline packages/rx-utils/src/__tests__/create-handler.utils.spec.ts]: Possibly unnecessary comments?

Comment (scink) [reply packages/rx-utils/src/__tests__/create-handler.utils.spec.ts]: no, i left it to have possibility to uncomment it and check that it fails as it should

Comment (raveclassic) [reply packages/rx-utils/src/__tests__/create-handler.utils.spec.ts]: comments are not checked :) 
either we should write correct test or remove it

Comment (scink) [reply packages/rx-utils/src/__tests__/create-handler.utils.spec.ts]: amended

**Pull request merged**